### ISSUE TITLE
Allow usage of secrets for PRs from forks

### ISSUE THREAD

**Issue by khaeru** [iiasa/message-ix-models#50]
For full requests from forks (e.g. #49), the "pytest" CI workflow currently fails. This is because it depends on secrets:
- GAMS_LICENSE for solving models.
- RTD_TOKEN_MESSAGE_DATA for retrieving the Intersphinx inventory. Missing this secret, however, only causes warnings, not a fatal error.

[GitHub does not pass secrets to PRs from forks](https://docs.github.com/en/actions/security-guides/encrypted-secrets). This is because any malicious third party could fork, write code that 'steals' the secret, and open a PR; then the automatically-triggered workflow would activate the malicious code.

It is _possible_ (we're not sure) that we could work around this using [GitHub's environments feature](https://docs.github.com/en/actions/using-jobs/using-environments-for-jobs). The idea is that, for each environment, we can set a maintainer (i.e. member of @iiasa/messageix-devs) who must give approval before particular workflows will run. Secrets can be stored within an environment (rather than for the repo as a whole).

To resolve this issue, investigate and, if viable, implement this solution.

Current workarounds:
- Core team members should make PRs from branches in this repo, rather than forks.
- For other PRs made from forks, use admin override to merge.

**Comment by khaeru** [comment]
We may also implement the same approach for message_data, although the concerns are less there because it is a private repository and we control who has access/is able to fork and open PRs.

**Comment by khaeru** [comment]
Sorry, maybe this is a duplicate of #21.

**Comment by jkikstra** [comment]
Thanks for opening this. Just commenting here to share that we've had a similar experience with another repository (climate assessment) with secrets, where we did not find a good solution (within time constraints), and just continued working using a branching workflow. 
So I'll keep an eye out to see how you progress on this.

**Comment by glatterf42** [comment]
This can likely be closed thanks to #281 and #288, too. @jkikstra, check out these PRs to see how we actually managed to enable PRs from forks to use our secrets :)

**Comment by khaeru** [comment]
Agreed, thanks for following up. Sorry also for duplicating #21 with this issue.
